module/core/src/Infrastructure/Validator/LanguageCodeValidator.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See license.txt for license details.
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\Core\Infrastructure\Validator;
+
+use Ergonode\Core\Domain\ValueObject\Language;
+use Ergonode\Core\Infrastructure\Validator\Constraint\LanguageCodeConstraint;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ */
+class LanguageCodeValidator extends ConstraintValidator
+{
+    /**
+     * @param mixed      $value
+     * @param Constraint $constraint
+     *
+     * @throws \ReflectionException
+     */
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof LanguageCodeConstraint) {
+            throw new UnexpectedTypeException($constraint, LanguageCodeConstraint::class);
+        }
+
+        if (empty($value)) {
+            return;
+        }
+        $languageReflection = new \ReflectionClass(Language::class);
+        $availableLanguages = $languageReflection->getConstants();
```

Comment:
Class Language have static method isValid its be easier to use it instead of reflection